app/src/components/ReleaseModal.tsx

```diff
@@ -0,0 +1,249 @@
+  const { data: resource } = trpc.draft.getDraftById.useQuery({
+    id: id,
+    resourceType: resourceType
+  });
+  const ctx = trpc.useContext();
+  const deleteMutation = trpc.draft.deleteDraft.useMutation({
+    onSuccess: () => {
+      notifications.show({
+        title: `Draft ${resource?.resourceType} released!`,
+        message: `Draft ${resource?.resourceType}/${resource?.id} successfully released to the MRS!`,
+        icon: <CircleCheck />,
+        color: 'green'
+      });
+      ctx.draft.getDraftCounts.invalidate();
+      ctx.draft.getDrafts.invalidate(); //? needed if redirecting?
+    },
+    onError: e => {
+      console.error(e);
+      notifications.show({
+        title: `Release Failed!`,
+        message: `Attempt to release ${resourceType} failed with message: ${e.message}`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    }
+  });
+
+  async function confirm() {
+    // requirements:
+    // https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#release
+    // TODO: release recursively all children (ignore for now).
+    // TODO/question: spec says don't change anything but status and date, but we're also changing version and idea (but spec is a mess)
+    if (resource) {
+      resource.version = version;
+      resource.status = 'active';
+      resource.date = DateTime.now().toISO() || '';
+    }
+
+    const res = await fetch(`${process.env.NEXT_PUBLIC_MRS_SERVER}/${resourceType}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json+fhir'
+      },
+      body: JSON.stringify(resource)
+    });
+
+    const location = res.headers.get('Location')?.substring(5); // remove 4_0_1 (version)
+    if (res.status !== 201) {
+      console.error(res.statusText);
+      notifications.show({
+        title: `Release Failed!`,
+        message: `Server unable to process request`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    } else if (!location) {
+      console.error('No resource location for released artifact');
+      notifications.show({
+        title: `Release Failed!`,
+        message: `No resource location exists for draft artifact`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    } else {
+      // delete draft
+      deleteMutation.mutate({
+        resourceType: resourceType,
+        id: id
+      });
+
+      //direct user to published artifact detail page
+      router.push(location);
+    }
+    //TODO/question: -> should we check service for whether artifact already exists in some way for PUT update?
+    onClose();
+  }
+
+  //This function is needed because, if I were to include this functionality where I call it,
+  // then it would appear on the modal which I don't want to happen
+  const setColor = (newColor: string) => {
+    currentColor = newColor;
+  };
+
+  return (
+    <Modal opened={open} onClose={onClose} withCloseButton={false} size="65%" centered>
```

Comment:
One thing I noticed which I should have brought up sooner is that sometimes UI/UX can be a pain when it comes to resizing the window. We don't need to worry about every window size, but it's good practice to enlargen or shrink your window to make sure the UI looks somewhat doable at different sizes. For example, this is what the modal current looks like when my window is a bit smaller than my laptop screen (which I think is a reasonable size for users:

<img width="1186" alt="image" src="https://github.com/projecttacoma/measure-repository/assets/30158156/bd6f9b1b-26ff-43c0-a7a8-9920eb2311c1">

I believe that the `size="65%"` here is partially the culprit, but in my experience, this requires some trial and error.

Reply:
Great catch! I didn't realize just how much I hard-coded sizing into the modal. I added some checks on the modal itself but not the list items and I was wondering if this helped your sizing in any way?